app/src/main/java/de/rampro/activitydiary/ui/history/HistoryActivity.java

```diff
@@ -176,44 +170,80 @@ protected void onCreate(Bundle savedInstanceState) {
-            if(data != null) {
+            if (data != null) {
                 long id = Long.decode(data.getLastPathSegment());
                 filterHistoryView(id);
             }
-        }else if (ActivityDiarySuggestionProvider.SEARCH_NOTE.equals(intent.getAction())) {
+        } else if (ActivityDiarySuggestionProvider.SEARCH_NOTE.equals(intent.getAction())) {
             Uri data = intent.getData();
-            if(data != null) {
-                String query = data.getLastPathSegment();
+            if (data != null) {
+                query = data.getLastPathSegment();
                 filterHistoryNotes(query);
             }
 
-        }else if (ActivityDiarySuggestionProvider.SEARCH_GLOBAL.equals(intent.getAction())) {
+        } else if (ActivityDiarySuggestionProvider.SEARCH_GLOBAL.equals(intent.getAction())) {
             Uri data = intent.getData();
-            if(data != null) {
-                String query = data.getLastPathSegment();
+            if (data != null) {
+                query = data.getLastPathSegment();
                 filterHistoryView(query);
             }
-        }else if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
-            String query = intent.getStringExtra(SearchManager.QUERY);
+        } else if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
+            query = intent.getStringExtra(SearchManager.QUERY);
             filterHistoryView(query);
         }
-            /* TODO: save recent query
-            SearchRecentSuggestions suggestions = new SearchRecentSuggestions(this,
-                    MySuggestionProvider.AUTHORITY, MySuggestionProvider.MODE);
-            suggestions.saveRecentQuery(query, null);
-*/
+
+        /*
+            if query was searched, then insert query into suggestion table
+         */
+        if (query != null) {
+            LocalDBHelper mOpenHelper = new LocalDBHelper(this);
+            SQLiteDatabase db = mOpenHelper.getWritableDatabase();
+            String sql = "INSERT INTO " + ActivityDiaryContract.DiarySuggestion.TABLE_NAME +
+                    "(" + ActivityDiaryContract.DiarySuggestion.SUGGESTION + ") " +
+                    "VALUES ('" + query + "');";
+
+            db.execSQL(sql);
+            long count = DatabaseUtils.queryNumEntries(db, ActivityDiaryContract.DiarySuggestion.TABLE_NAME);
+            /*if table contains more than 5 suggestions, then remove the oldest one*/
+            if (count > 5) {
+                sql = "DELETE FROM " + ActivityDiaryContract.DiarySuggestion.TABLE_NAME +
+                        " WHERE " + ActivityDiaryContract.DiarySuggestion.lAST_CHANGED +
+                        " IN (SELECT " + ActivityDiaryContract.DiarySuggestion.lAST_CHANGED +
+                        " FROM " + ActivityDiaryContract.DiarySuggestion.TABLE_NAME +
+                        " ORDER BY " + ActivityDiaryContract.DiarySuggestion.lAST_CHANGED + " ASC LIMIT 1);";
+                db.execSQL(sql);
```

Comment:
What happens in case the user clicks on a recent searches suggestion? Is it readded and we can have the same search in slot 3 and 1? I'd propose to remove it in the location where it sits and readd it at the beginning. (This could be a reason to not use the AUTOINCREMENT on a primary key for sorting, but update the ID manually to be able to resort easily (and not having to check whether you can also manually overwrite an autoincrement field by an explicit new number)